src/components/home/HomeActions/index.tsx

```diff
@@ -0,0 +1,50 @@
+import { Typography } from '@/components/common';
+import { UserProgress } from '@/components/profile/UserProgress';
+import { PrivateProfile } from '@/lib/types/apiResponses';
+import CheckMark from '@/public/assets/icons/check-mark.svg';
+import { FormEvent, useState } from 'react';
+import styles from './style.module.scss';
+
+interface HomeActionsProps {
+  user: PrivateProfile;
+  points: number;
+  checkin: (code: string) => void;
+}
+
+const HomeActions = ({ user, points, checkin }: HomeActionsProps) => {
+  const [checkinCode, setCheckinCode] = useState('');
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (checkinCode !== '') {
+      checkin(checkinCode);
```

Comment:
should it reset the text in the check-in code input when you successfully check in

Reply:
Sure